### PR TITLE
OCLOMRS-575: Allow a user to search for concepts from internal sources while creating mappings

### PR DESCRIPTION
**Pull request by Karuhanga**
# JIRA TICKET NAME:
[Allow a user to search for concepts from internal sources while creating mappings](https://issues.openmrs.org/browse/OCLOMRS-575)

# Summary:
When creating mappings, a user should be able to search for concepts from internal sources that they might want to select

### CODE REVIEW

File: src/tests/dictionaryConcepts/components/CreateMapping.test.js [hunk 1]
```diff
@@ -256,4 +314,38 @@ describe('Test suite for dictionary concepts components', () => {
     await inputField.instance().handleKeyPress(event, 'malaria');
     expect(spy).toHaveBeenCalled();
   });
+
+  describe('selectInternalMapping', () => {
+    it('should notify a user when they try to search with less than three characters', async () => {
```
Comment (dkayiwa) [inline src/tests/dictionaryConcepts/components/CreateMapping.test.js]: Where are the less than three characters passed?

Comment (Karuhanga) [reply src/tests/dictionaryConcepts/components/CreateMapping.test.js]: I somehow mixed up this description. Thanks for the catch.

**Comment by Karuhanga** [comment]
The unrelated changes [in this test file](https://github.com/openmrs/openmrs-ocl-client/pull/456/files#diff-953833039f51d0296ff9e48a950338b8) were because a missing mock was causing some tests to fail when that warning was thrown.

File: src/tests/dictionaryConcepts/components/CreateMapping.test.js [hunk 2]
```diff
@@ -170,6 +194,102 @@ describe('Test suite for dictionary concepts components', () => {
...
+    const createMappingInstance = wrapper.find('CreateMapping').instance();
+
+    expect(notifyMock).not.toHaveBeenCalled();
+
+    await createMappingInstance.handleKeyPress(event, inputValue, url, false);
+
+    expect(notifyMock).toHaveBeenCalledWith('Query must have at least three characters', 'warning', 2000);
+  });
+
+  it('should display a message when there are no concepts to select from', (done) => {
```
Comment (dkayiwa) [inline src/tests/dictionaryConcepts/components/CreateMapping.test.js]: Which message?

Comment (Karuhanga) [reply src/tests/dictionaryConcepts/components/CreateMapping.test.js]: Updated this to be more explicit.

**Comment by dkayiwa** [comment]
Some merge conflicts here.

**Comment by Karuhanga** [comment]
> Some merge conflicts here.

Addressed.

**Comment by Karuhanga** [comment]
> Did you see the travis build failure?

Seen this. Looking into it.

**Comment by Karuhanga** [comment]
> Did you see the travis build failure?

This has now been addressed.

**Pull request merged**